### PR TITLE
feat: projectname param includes product hierarchy + guides

### PR DESCRIPTION
**Pull request by eaball35**
Optional `-projectname` param includes product hierarchy and guides (Overview, Version history) in toc

Fixes b/179793720, b/179920619

### CODE REVIEW

File: third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package com.microsoft.model;
```
Comment (lesv) [inline third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java:1]: Copyright & license

File: third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        assertThat("Guide should be first", items.get(0), instanceOf(Guide.class));
+        Guide overview = (Guide) items.get(0);
+        assertThat("First guide should be Overview", overview.getName(), is("Overview"));
+        assertThat("Guide should be second", items.get(1), instanceOf(Guide.class));
+        Guide history = (Guide) items.get(1);
+        assertThat("Second guide should be Version History", history.getName(), is("Version history"));
+
+        assertThat("Item A should be second", items.get(2), is(tocItemA));
+        assertThat("Item B should be third", items.get(3), is(tocItemB));
+        assertThat("Item C should be fourth", items.get(4), is(tocItemC));
```
Comment (lesv) [inline third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java:56]: Consider using Truth in the future.  (Fluent API)

File: third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class TocContentsTest {
```
Comment (lesv) [inline third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java:13]: Are there new generated files that should be checked?

**Review by lesv — CHANGES_REQUESTED** [review]

File: third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java (hunk 2 above)
Comment (eaball35) [reply third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java:72]: Will do

File: third_party/docfx-doclet-143274/src/main/java/com/microsoft/model/ProjectContents.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        this.name = name;
+        this.items = items;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public List<Object> getItems() {
+        return items;
```
[the remark this answers is not in the document]
Comment (eaball35) [reply third_party/docfx-doclet-143274/src/main/java/com/microsoft/model/ProjectContents.java:37]: Makes sense

File: third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java (hunk 3 above)
Comment (eaball35) [reply third_party/docfx-doclet-143274/src/test/java/com/microsoft/model/TocContentsTest.java:29]: This test just shows that toc contents is structuring the content as intended. For generated files, I updated the previous tests to include the new param and changed the expected outputs. Updates from this to toc will look like [expected-generated-files/toc.yml](https://github.com/googleapis/java-docfx-doclet/blob/dc6aafae1d812de80b3b4ed2b28909e36a9fa3c0/third_party/docfx-doclet-143274/src/test/resources/expected-generated-files/toc.yml)

File: third_party/docfx-doclet-143274/src/main/java/com/microsoft/model/TocFile.java [hunk 5]
```diff
@@ -3,15 +3,18 @@
...
 
 public class TocFile extends ArrayList<TocItem> implements YmlFile {
 
     private final static String TOC_FILE_HEADER = "### YamlMime:TableOfContent\n";
     private final static String TOC_FILE_NAME = "toc.yml";
     private final String outputPath;
+    private final String projectName;
 
-    public TocFile(String outputPath) {
+    public TocFile(String outputPath, String projectName) {
```
[the remark this answers is not in the document]
Comment (eaball35) [reply third_party/docfx-doclet-143274/src/main/java/com/microsoft/model/TocFile.java:15]: I think I'll leave it as is for now, but will reconsider this as I refactor in the future

**Review by lesv — APPROVED** [review]

**Pull request merged**